ddtrace/internal/writer.py

```diff
@@ -131,7 +154,12 @@ def recreate(self):
         writer = self.__class__(out=self.out, sampler=self._sampler, priority_sampler=self._priority_sampler)
         return writer
 
-    def write(self, spans=None):
+    def stop(self, timeout=None):
+        # type: (Optional[float]) -> None
+        return
+
+    def write(self, spans):
```

Comment:
Hi @Kyle-Verhoog , long time no see!  Just kidding this is Patrick from today's Instacart call. Hey here is another backwards incompatibility I just found getting us on the latest.  `spans=None`.  We had to update 1 reference to LogWriter.write() and change it to `.write(none)` to accommodate the removal fo the default None for `spans`.  Not a big deal but pointing another one out for you ;) thanks again

Reply:
Hey Patrick, very interesting! I'm curious as to why you were using a `LogWriter` manually 🤔 . Could you share a bit more about your usage as it might be something we should have support for in the public API.

It is in our `internal` module so I think it was fair-game but definitely appreciate the heads-up! 🙂

Reply:
So I disagree with that.  Although `LogWriter` is defined in your internal subpackage, we are calling it publicly: `tracer.writer.writer()` from `ddtrace.tracer`.  

`ddtrace.__init__` instantiates a public global tracer:
```
# a global tracer instance with integration settings
tracer = Tracer()
```

`writer` is a public member on `Tracer` which is an a instantiation of `LogWriter`, and `LogWriter.write()` is a public method.  

So I get that you are putting your class definition in a "private" subpackage, however since you are making a global instance of that public via a chain of public members ultimately made public at the top level, it is not fair game as you say.  I don't think it's correct procedure for developers to have to inspect every public API member to see where the class definition is before using it.  If it is a public member or method, that is good enough.  If `Tracer.writer` was actually `Tracer._writer` then you would have a fair case that we are violating the public/private API, but this is no the case!

As far as what we are doing with it, we are calling it this way after each `rq` job is done processing.  I don't know the background here or if this is the optimal way to accomplish what we are going for, i'll have to check with the more seasoned team members and get back to you:
```
        # necessary to manually flush traces due to rq's forking model
        tracer.writer.write()
        tracer.writer.stop()
        tracer.writer.flush_queue()
```

Reply:
Ahh yes, fair enough! Totally agreed with that level of access that the writer falls into the public API. The fault here is that we expose `writer` publicly (we're working to do away with this) on tracer and there should be an API on tracer to flush the traces.

> I don't think it's correct procedure for developers to have to inspect every public API member to see where the class definition is before using it.

100% agreed :+1: I was naive in thinking that your usage was by importing and using the `LogWriter` directly.

Reply:
@DataDog/apm-python this is additional support to add to our 1.0 discussions that it's really hard to not make subtle breaking changes with the current API as we're working towards the API we want (almost everything in our internal module is exposed in some way through the public API).